lib/segment/src/segment.rs

```diff
@@ -766,6 +806,25 @@ impl SegmentEntry for Segment {
         }
     }
 
+    fn update_vectors(
```

Comment:
It is probably good to look into this function with extra care.

Reply:
lgtm. I would expect it to be exactly as update of payload